front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/HeaderImageDropzone.tsx

```diff
@@ -89,19 +90,26 @@ const HeaderImageDropzone = ({
     return ratio;
   };
 
-  const theme = useTheme();
-  const displayOverlayColor = overlayColor ?? theme.colors.tenantPrimary;
-  const displayOverlayOpacity =
-    overlayOpacity ?? theme.signedOutHeaderOverlayOpacity;
+  // Ideally, this condition is not here. On changing the layout,
+  // We should set the color/opacity to null. This should be refactored
+  // at the next convenience.
+  const isBannerWithOverlay =
+    layout === 'full_width_banner_layout' || layout === 'fixed_ratio_layout';
 
   const previewOverlayElement =
-    (layout === 'full_width_banner_layout' ||
-      layout === 'fixed_ratio_layout') &&
-    displayOverlayColor &&
-    displayOverlayOpacity ? (
+    /*
+      We check for typeof of opacity because 0 would coerce to false.
+      We don't do a similar thing for the color because we don't set
+      it in the toggle hander, so we handle the type check with the
+      theme default as fallback
+    */
+    isBannerWithOverlay && typeof overlayOpacity === 'number' ? (
```

Comment:
Maybe the same as before would make sense

```suggestion
    isBannerWithOverlay && overlayOpacity !== null ? (
```